index.js

```diff
@@ -156,6 +154,10 @@ class AddLambdaInsights {
    * @return {Promise} Lambda Insights Layer ARN
    */
   addLambdaInsights() {
+    // resolve provider and region properties
+    this.provider = this.serverless.getProvider('aws');
+    this.region = this.provider.getRegion();
```

Comment:
Do we need to store this as state on the plug-in? Or would it make sense to just pass it down from here? Or just call getProvider directly whenever we need it? I’m not familiar with the lifecycle of a Serverless plug-in so it’s hard to weigh in on the right option.

Reply:
you are probably right - getRegion() can be called when needed
I will change that and merge